embeddings/data/datamodule.py

```diff
@@ -0,0 +1,167 @@
+import abc
+from typing import Any, Dict, Generic, List, Optional, TypeVar, Union
+
+import datasets
+import pytorch_lightning as pl
+from datasets import DatasetDict
+from torch.utils.data import DataLoader
+from transformers import AutoTokenizer, BatchEncoding
+
+from embeddings.utils.loggers import get_logger
+
+Data = TypeVar("Data")
+HuggingFaceDataset = TypeVar("HuggingFaceDataset")
+
+_logger = get_logger(__name__)
+
+
+class BaseDataModule(abc.ABC, pl.LightningDataModule, Generic[Data]):
+    dataset: Data
+
+
+class HuggingFaceDataModule(BaseDataModule[DatasetDict]):
+    loader_columns = [
+        "datasets_idx",
+        "input_ids",
+        "token_type_ids",
+        "attention_mask",
+        "start_positions",
+        "end_positions",
+        "labels",
+    ]
+    num_labels: int
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        dataset_name: str,
+        target_field: str,
+        max_seq_length: int = 128,
+        train_batch_size: int = 32,
+        eval_batch_size: int = 32,
+        input_dim: int = 768,
+        load_dataset_kwargs: Optional[Dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__()  # type: ignore
+        self.model_name_or_path = model_name_or_path
+        self.dataset_name = dataset_name
+        self.target_field = target_field
+        self.max_seq_length = max_seq_length
+        self.train_batch_size = train_batch_size
+        self.eval_batch_size = eval_batch_size
+        self.input_dim = input_dim
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name_or_path, use_fast=True)
+        self.initialized = False
+        if load_dataset_kwargs is None:
+            self.load_dataset_kwargs = {}
+        else:
+            self.load_dataset_kwargs = load_dataset_kwargs
+
+    @property
+    def output_dim(self) -> Optional[int]:
+        if not self.initialized:
+            _logger.warning("Datamodule not initialized. Returning None.")
+            return None
+        else:
+            return self.num_labels
+
+    def prepare_data(self) -> None:
+        datasets.load_dataset(self.dataset_name)
+        AutoTokenizer.from_pretrained(self.model_name_or_path, use_fast=True)
+
+    def setup(self, stage: Optional[str] = None) -> None:
+        if not self.initialized:
+            self.initalize()
+        self.process_data(stage)
+
+    def process_data(self, stage: Optional[str] = None) -> None:
+        if stage == "fit":
```

Comment:
Is only `fit` needed here?

Reply:
changed to "fit" or none; 

from https://pytorch-lightning.readthedocs.io/en/latest/notebooks/lightning_examples/datamodules.html
![image](https://user-images.githubusercontent.com/26749468/145418695-4fbd6aa0-d9c5-4650-898f-74b7c83d82ba.png)